### PR TITLE
importer: patch few things along the way

### PR DESCRIPTION
**Pull request by arturtamborski**
#### Story / Bug id:
4

#### Description:
Download csv from Gdrive and put into backend/temp folder. It's one-time thing. 

#### Migrations:
N/A

#### New imports / dependencies:
N/A

#### What tests do I need to run to validate this change:
N/A

### CODE REVIEW

File: backend/commands/import_attendance_list.py [hunk 1]
```diff
@@ -30,24 +30,30 @@ def import_attendance_list():
         users_dict = {}
         _id, github, name, phone, email, *_ = item
 
-        first_name = name.split()[0]
         try:
+            first_name = name.split()[0]
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:34]: this sometimes failed

File: backend/commands/import_attendance_list.py [hunk 2]
```diff
@@ -30,24 +30,30 @@ def import_attendance_list():
...
             last_name = name.split()[1]
         except:
+            first_name = "Unknown"
             last_name = "Unknown"
 
-        users_dict["id"] = _id
+        # skip when _id == ''
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:40]: no idea why exactly that happened, but still - that fixed it :P

File: backend/commands/import_attendance_list.py [hunk 3]
```diff
@@ -30,24 +30,30 @@ def import_attendance_list():
...
         users_dict["email"] = email
         users_dict["phone"] = phone
         list_of_participants.append(users_dict)
 
+
     # create hacknights
-    for date in dates:
-        year, month, day = date.split(".")
-        hacknight = Hacknight(date=f"{year}-{month}-{day}")
+    for idx, date in enumerate(dates, 1):
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:54]: Hacknight() requires passing `id=` as well, that's a simple way of getting it

File: backend/commands/import_attendance_list.py [hunk 4]
```diff
@@ -30,24 +30,30 @@ def import_attendance_list():
...
+        day, month, year = date.split(".")
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:55]: turns out these two were swapped

File: backend/commands/import_attendance_list.py [hunk 5]
```diff
@@ -65,10 +71,13 @@ def import_attendance_list():
 
     # assign attendance list
     for participant_idx, row in enumerate(rows, 1):
+        participant = Participant.query.get(participant_idx)
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:74]: it's faster that way

File: backend/commands/import_attendance_list.py [hunk 6]
```diff
@@ -65,10 +71,13 @@ def import_attendance_list():
...
         for hacknight_index, element in enumerate(row[5:], 1):
-            participant = Participant.query.get(participant_idx)
-            if "1" in element:
+            if "1" in element and participant:
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:76]: sometimes `participant` is completely empty, no idea why, I just ignored such cases

File: backend/commands/import_attendance_list.py [hunk 7]
```diff
@@ -65,10 +71,13 @@ def import_attendance_list():
...
                 hacknight = Hacknight.query.get(hacknight_index)
                 hacknight.participants.append(participant)
-                db.session.commit()
-                click.echo(f"Added {participant.github} to hacknight {hacknight.date}")
+                email = participant.email if participant else '???'
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:79]: turns out, email _can_ be empty. lazy replacement is used in such case

File: backend/commands/import_attendance_list.py [hunk 8]
```diff
@@ -65,10 +71,13 @@ def import_attendance_list():
...
+                click.echo(f"Added {email} to hacknight {hacknight.date}")
+        db.session.flush()
```
Comment (arturtamborski) [inline backend/commands/import_attendance_list.py:81]: again, it's bit faster to flush and then commit()

**Review by w1stler — APPROVED** [review]

**Pull request merged**